### PR TITLE
Feature - Final Project - Implement Base Moya

### PR DESCRIPTION
**Pull request by mvn-phonghuynh-dn**

### CODE REVIEW

File: RxSwift_Template/Models/Core/ApiNetWorkManager.swift [hunk 1]
```diff
@@ -13,32 +13,53 @@ final class ApiNetWorkManager {
...
-                        single(.failure(error))
+                        throw ApiError.noData
                     }
-                case .failure(let error):
-                    single(.failure(error))
+                case 400...499:
+                    throw ApiError.badRequest
+                default:
+                    throw ApiError.unknown
                 }
```
Comment (mvn-quocdoan-dn) [inline RxSwift_Template/Models/Core/ApiNetWorkManager.swift:36]: Chuyên nghiệp lên nào

File: RxSwift_Template/Models/Core/ApiNetWorkManager.swift [hunk 2]
```diff
@@ -13,32 +13,66 @@ final class ApiNetWorkManager {
...
+                do {
+                    return try JSONDecoder().decode(T.self, from: response.data)
+                } catch {
+                    throw ApiError.noData
                 }
             }
-            return Disposables.create()
+            .catch { error in
+                throw error
+            }
```
Comment (mvn-quocdoan-dn) [inline RxSwift_Template/Models/Core/ApiNetWorkManager.swift:34]: Đồng nhất type of error, please

**Review by mvn-annguyen-dn — CHANGES_REQUESTED** [review]
- separate code by particular functionalities
- when u make task that is related to UI, attach screenshot UI
@mvn-phonghuynh-dn  please fix my comments, Thanks

File: RxSwift_Template/AppDelegate/AppDelegate.swift [hunk 3]
```diff
@@ -9,16 +9,27 @@ import UIKit
...
+
+    static var shared: AppDelegate = {
+        guard let delegate = UIApplication.shared.delegate as? AppDelegate else {
+            fatalError("Can not get AppDelegate")
+        }
+        return delegate
+    }()
+
     var window: UIWindow?
+    let tabbarController = UITabBarController()
```
Comment (mvn-quocdoan-dn) [inline RxSwift_Template/AppDelegate/AppDelegate.swift:26]: Remove please

File: RxSwift_Template/Models/Core/LoginTarget.swift [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+enum LoginTarget {
+    case example
+}
+
+extension LoginTarget: TargetType {
+    
+    var baseURL: URL {
+        guard let baseUrl: URL = URL(string: "https://rss.applemarketingtools.com/api/v2/us/music/") else {
+            fatalError(ApiError.unknown.localizedDescription)
```
Comment (mvn-annguyen-dn) [inline RxSwift_Template/Models/Core/LoginTarget.swift:18]: shouldn't use fatalError => throw

File: RxSwift_Template/Models/Scheme/Example.swift [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+     var results: [Music]?
+
+     init(results: [Music]?) {
+         self.results = results
+     }
+
+     enum CodingKeys: String, CodingKey {
+         case feed
+         case results
+     }
```
Comment (mvn-quocdoan-dn) [inline RxSwift_Template/Models/Scheme/Example.swift:57]: ??

File: RxSwift_Template/Models/Scheme/Example.swift [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+     enum CodingKeys: String, CodingKey {
+         case artistName
+         case id
+         case name
+         case releaseDate
+         case copyright
+         case artworkUrl100
+     }
```
Comment (mvn-quocdoan-dn) [inline RxSwift_Template/Models/Scheme/Example.swift:34]: ??

**Review by mvn-quocdoan-dn — COMMENTED** [review]
@mvn-luongtran-dn Hi luong, please check my comment. Thanks!

**Review by mvn-annguyen-dn — CHANGES_REQUESTED** [review]
- Create BaseViewController is made common function (for instance: showAlertError, handleSomething) for all of viewcontroller
- Check my comments, Thanks
@mvn-luongtran-dn

**Pull request merged**